### PR TITLE
Fix: Viewport width not accounted for in the Site Editor > Patterns view

### LINKED ISSUE #55324: Viewport width not accounted for in the site editor > patterns view
Twenty Twenty-Four includes a number of patterns that set the viewport width of the pattern ([source](https://github.com/WordPress/twentytwentyfour/blob/trunk/patterns/clients.php#L7)), so that a pattern display nicely within the inserter. 

This works fine in the inserter, but within the site editor > patterns view, the viewport width value is omitted. I would expect the patterns to display exactly the same anywhere the preview is rendered. 

As seen in the patterns inserter and explorer (expected):
CleanShot 2023-10-12 at 15 09 36
CleanShot 2023-10-12 at 15 11 56

As seen in the site editor > patterns view: 
CleanShot 2023-10-12 at 15 09 44

### PR DESCRIPTION
**Pull request by t-hamano**
Fixes #55324

## What?

This PR fixes an issue where the Site Editor's Pattern View does not take into account the Viewport Width defined in the pattern.

### Before

before

### After 

after

## Why?

The `BlockPreview` component that is rendering the pattern receives a `viewportWidth` prop to change the viewport width.
However, in the Site Editor's Pattern preview, the `viewportWidth` property is not passed, so it is displayed with a width of `1200px`, which is the default viewport width value of the `BlockPreview` component.

## How?

I passed the pattern's `viewportWidth` as a property to the `BlockPreview` component.

## Testing Instructions

For Emptytheme, create two patterns with defined viewport widths:

<details><summary>test/emptytheme/patterns/width-500.php</summary>

```html
<?php
/**
 * Title: Viewport Width 500
 * Slug: emptytheme/width-500
 * Categories: banner
 * Viewport Width: 500
 */
?>

<!-- wp:paragraph {"style":{"typography":{"fontSize":"60px"}}} -->
<p style="font-size:60px">ABCDEFGHIJKLMNOPQRSTUVWXYZABCDEFGHIJKLMNOPQRSTUVWXYZ</p>
<!-- /wp:paragraph -->
```

</details> 

<details><summary>test/emptytheme/patterns/width-1000.php</summary>

```html
<?php
/**
 * Title: Viewport Width 1000
 * Slug: emptytheme/width-1000
 * Categories: banner
 * Viewport Width: 1000
 */
?>

<!-- wp:paragraph {"style":{"typography":{"fontSize":"60px"}}} -->
<p style="font-size:60px">ABCDEFGHIJKLMNOPQRSTUVWXYZABCDEFGHIJKLMNOPQRSTUVWXYZ</p>
<!-- /wp:paragraph -->
```

</details> 

### Testing Instructions for Keyboard

- Activate Emptytheme.
- Go to the Site Editor > Patterns.
- Both patterns should have the correct viewport width (you can see this in the width style applied to the iframe element in the pattern).
- Patterns that do not have a viewport width, such as those created in the Site Editor, should still have a default viewport width of 1200px.

### CODE REVIEW

File: packages/edit-site/src/components/page-patterns/grid-item.js [hunk 1]
```diff
@@ -200,6 +200,7 @@ function GridItem( { categoryId, item, ...props } ) {
 					<BlockPreview
 						blocks={ item.blocks }
 						additionalStyles={ additionalStyles }
+						viewportWidth={ item?.viewportWidth }
```
Comment (aaronrobertshaw) [inline packages/edit-site/src/components/page-patterns/grid-item.js]: Nit: I don't think we need the optional chaining here

**Review by aaronrobertshaw — APPROVED** [review]
Thanks for addressing this @t-hamano 👍 

It tests well for me. 

| Before | After |
|---|---|
| Screenshot 2023-11-06 at 11 27 02 am | Screenshot 2023-11-06 at 11 26 25 am | 

IIRC there were some issues with the block previews when responsive units could be used, it would be great to get a further sanity check from @kevin940726 on this front.

File: packages/edit-site/src/components/page-patterns/grid-item.js (hunk 1 above)
Comment (t-hamano) [reply packages/edit-site/src/components/page-patterns/grid-item.js]: Indeed 👍

**Comment by t-hamano** [comment]
Thanks for the review, @aaronrobertshaw!

> IIRC there were some issues with the block previews when responsive units could be used, it would be great to get a further sanity check from @kevin940726 on this front.

Is there anything I can do about this? The `viewPortWidth` property of the `BlockPreview` component is already used in many places, so if there is a problem, I think this could be addressed as a separate PR.

**Comment by aaronrobertshaw** [comment]
> I think this could be addressed as a separate PR.

Sounds like a plan 👍

**Comment by t-hamano** [comment]
Thanks for the reply, @aaronrobertshaw! I would like to rebase this PR with the latest trunk and merge it if there are no problems.

**Pull request merged**